consistent-sampling/src/main/java/io/opentelemetry/contrib/samplers/ConsistentRateLimitingSampler.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.contrib.samplers;
+
+import static java.util.Objects.requireNonNull;
+
+import io.opentelemetry.contrib.util.DefaultRandomGenerator;
+import io.opentelemetry.contrib.util.RandomGenerator;
+import io.opentelemetry.sdk.trace.samplers.Sampler;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.LongSupplier;
+import javax.annotation.concurrent.Immutable;
+
+/**
+ * This consistent {@link Sampler} adjust the sampling probability dynamically to limit the rate of
+ * sampled spans.
+ *
+ * <p>This sampler uses exponential smoothing to estimate on irregular data (compare Wright, David
+ * J. "Forecasting data published at irregular time intervals using an extension of Holt's method."
+ * Management science 32.4 (1986): 499-510.) to estimate the current rate of spans.
```

Comment:
Looking at Wright's method (equations 2.5, 2.7 [here](http://quantitative.cz/wp-content/uploads/2018/09/improved-holt-method-for-irregular-time-series-tomas-hanzak-2008.pdf)), there seem to be some differences from what you've implemented. E.g., there are no `1 - decayFactor` terms in this implementation, whereas 1 - alpha appears in Wright. Is this an intentional deviation?

Maybe because the setting in OTel sampling is just different than that of forecasting? My understanding of this implementation is: we're trying to estimate a rate, so you need to maintain both an exponentially smoothed span count and an exponentially smoothed window duration in order for their ratio to be a meaningful: an exponential moving _rate_ (in analogy to exponential moving average). Remarkably, I can't find any literature on "exponential moving rate".